app/src/main/java/me/quadphase/qpdex/databaseAccess/PokemonFactory.java

```diff
@@ -1122,6 +1196,79 @@ public void loadAllTypes() {
     }
 
     /**
+     * Loads all the Abilities into memory. Should be done at program start (during splash screen)
+     */
+    public void loadAllAbilities() {
+        // fail safe 0th index of types:
+        abilities[0] = new Ability("None","None");
+        // get the real abilities from the database:
+        for (int i = 1; i < getMaxAbilityID() + 1; i++) {
+            String[] selectionArg = {String.valueOf(i)};
+            Cursor cursor = database.query(ABILITIES_TABLE, null, ABILITIY_ID + "=?", selectionArg, null, null, null);
+            cursor.moveToFirst();
+
+            abilities[i] = new Ability(cursor.getString(cursor.getColumnIndex(NAME)), cursor.getString(cursor.getColumnIndex(DESCRIPTION)));
+
+            // close the cursor
+            cursor.close();
+        }
+    }
+
+    /**
+     * Loads all the EggGroups into memory. Should be done at program start (during splash screen)
+     */
+    public void loadAllEggGroups() {
+        // fail safe 0th index of types:
+        eggGroups[0] = new EggGroup("None");
+        // get the real abilities from the database:
+        for (int i = 1; i < getMaxEggGroupID() + 1; i++) {
+            // eggGroupId's start at 0
+            String[] selectionArg = {String.valueOf(i - 1)};
+            Cursor cursor = database.query(EGG_GROUPS_TABLE, null, EGG_GROUP_ID + "=?", selectionArg, null, null, null);
+            cursor.moveToFirst();
+
+            eggGroups[i] = new EggGroup(cursor.getString(cursor.getColumnIndex(NAME)));
```

Comment:
Consider adding the eggGroupID as a private variable to the EggGroup class (like with Type). This will give easier access to the index in the array.